bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,956 @@
+                    }
+                    break;
+                default:
+                    logger.warn(
+                            "air-Q - airqHandler - handleCommand(): unknown command {} received (channelUID={}, value={})",
+                            command, channelUID, command);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getThing().getConfiguration().as(AirqConfiguration.class);
+        logger.debug("air-Q - airqHandler - initialize(): config={}", config);
+        updateStatus(ThingStatus.UNKNOWN);
+        if ((config.ipAddress.isEmpty()) || (config.password.isEmpty())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "IP Address and the device password must be provided to access your air-Q.");
+            return;
+        } else {
+            // we try if the device is reachable and the password is correct. Otherwise a corresponding message is
+            // thrown in Thing manager.
+            String data = getDecryptedContentString("http://".concat(config.ipAddress.concat("/data")), "GET", null);
+            if (data == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "We tried to get data from the air-Q device, but failed. Probable cause: invalid password.");
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+        }
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollData, 0, POLLING_PERIOD_DATA, TimeUnit.MILLISECONDS);
+        getConfigDataJob = scheduler.scheduleWithFixedDelay(this::getConfigData, 0, POLLING_PERIOD_CONFIG,
+                TimeUnit.MINUTES);
+        logger.debug("air-Q - airqHandler - initialize() finished");
+    }
+
+    // AES decoding based on this tutorial: https://www.javainterviewpoint.com/aes-256-encryption-and-decryption/
+    public @Nullable String decrypt(byte[] base64text, String password) {
+        String content = "";
+        logger.trace("air-Q - airqHandler - decrypt(): content to decypt: {}", base64text);
+        byte[] encodedtextwithIV = Base64.getDecoder().decode(base64text);
+        byte[] ciphertext = Arrays.copyOfRange(encodedtextwithIV, 16, encodedtextwithIV.length);
+        byte[] passkey = Arrays.copyOf(password.getBytes(), 32);
+        if (password.length() < 32) {
+            Arrays.fill(passkey, password.length(), 32, (byte) '0');
+        }
+        byte[] IV = Arrays.copyOf(encodedtextwithIV, 16);
+        // We comment this out because by setting the appropriate log level a third party could find the air-Q password.
+        // logger.trace("air-Q - airqHandler - decrypt(): passkey={}", passkey);
+        // logger.trace("air-Q - airqHandler - decrypt(): IV={}", IV);
+        // logger.trace("air-Q - airqHandler - decrypt(): text to decode: {}", ciphertext);
```

Comment:
a general rule is not to merge commented out code, any code that is commented out should be removed, if it is needed to help people that may hit an issue then leave it as TRACE or DEBUG. Lastly just for knowledge if you have multiple logger calls like this in a row, you should test to see if TRACE is enabled in an if statement so the test is only done once and not three times.
Also agree to try and keep passwords out of the log, actually the same thought for base64 encoded material. It is very easy to reverse the encoding that it offers no protection at all in my opinion.